File: myproj26/main13-map-function.js
const numbers = [1, 3, 5, 4, 2];

// Array의 sort
function make_random_value(number1, number2) {
    return Math.random();
}

// -- 체이닝 --
const new_numbers = numbers.map(
    (number) => ({ number, 기준값: Math.random() }),
)
// .sort(
//     (value1, value2) => {
//         return value1.기준값 - value2.기준값;
//     }
// ).map(
//     (value) => value.number
// );

console.log(new_numbers);

// js 에서 sort는 비교함수 --> 값을 두 개 받아야 함



// 새로운 Array(배열), 제곱수로 구성된 숫자 배열
// function mapper(number) {
//     return number ** 2;
// }

// const mapper = (number) => {
//     return number ** 2;
// }

// const mapper = (number) => number ** 2;

// const mapper = number => number ** 2;

// const new_numbers = numbers.map(mapper);
// console.log(new_numbers);

// const new_numbers = numbers.map(
//     number => number ** 2
// );
// console.log(new_numbers);

File: myproj26/main13-map-function.py
import random

numbers = [31, 89, 24, 81, 46]

# 31, 81, 24, 46, 89  # ASC (1의 자리 기준)
#  1   1   4   6   9
# 1) 내장함수 sorted
print(sorted(numbers))  # 오름차순 정렬
print(sorted(numbers, reverse=True))  # 내림차순 정렬


def make_value(number):
    return number % 10


print(sorted(numbers, key=make_value))

# 2) 리스트의 sort
# numbers.sort()
# numbers.sort(key=make_value)
# print(numbers)

# numbers.sort(key=lambda number: number % 10)
# print(numbers)


def make_random_value(number):
    return random.randint(1, 100)


numbers.sort(key=make_random_value)
print("random values :", numbers)

# -----

# # 내장함수 map
# # def mapper(number):
# #     return number ** 2
# # print(list(map(mapper, numbers)))

# print(list(map(
#     lambda number: number ** 2,
#     numbers,
# )))

# # new_numbers = []
# # for number in numbers:
# #     new_numbers.append(number ** 2)

# # # list comprehension
# # new_numbers = [
# #     number ** 2 for number in numbers
# # ]
